src/dataviews/category-dataview/categories-collection.js

```diff
@@ -10,6 +10,27 @@ var CategoryItemModel = require('./category-item-model');
 module.exports = Backbone.Collection.extend({
   model: CategoryItemModel,
 
+  initialize: function (models, options) {
+    this.aggregationModel = options.aggregationModel;
+    this.aggregation = options.aggregationModel.get('aggregation');
+    this.aggregationModel.on('change:aggregation', function (model, aggregation) {
+      this.aggregation = aggregation;
+      this.filterNull();
+    }, this);
+
+    this.on('reset', this.filterNull, this);
+  },
+
+  filterNull: function () {
```

Comment:
Another option could be to override the `reset` method and filter out categories with `null` values. eg:

```
  reset: function(models, options) {
    if (this.aggregation !== COUNT_AGGREGATION_TYPE) {
      models = _.filter(models, function (category) {
        return category.get('value') != null;
      });
      Backbone.Collection.prototype.reset.call(models, options);
    }
  }
```

```